File: packages/rust/armonik/src/client/versions.rs
pub use crate::rpc::versions::Client as Versions;

use crate::client::client_method;

#[armonik_macros::client]
#[armonik(service = "armonik.api.grpc.v1.versions.Versions")]
impl<T: super::Channel> super::ServiceClient<crate::rpc::services::Versions, T> {
    client_method!(ListVersions:
        list()
        -> crate::versions::list::Request => crate::versions::list::Response);
}

File: packages/python/src/armonik/client/versions.py
from typing import Dict

from grpc import Channel

from ..protogen.client.versions_service_pb2_grpc import VersionsStub
from ..protogen.common.versions_common_pb2 import (
    ListVersionsRequest,
    ListVersionsResponse,
)


class ArmoniKVersions:
    def __init__(self, grpc_channel: Channel):
        """Result service client

        Args:
            grpc_channel: gRPC channel to use
        """
        self._client = VersionsStub(grpc_channel)

    def list_versions(self) -> Dict[str, str]:
        """Get versions of ArmoniK components.

        Return:
            A dictionnary mapping each component to its version.
        """
        request = ListVersionsRequest()
        response: ListVersionsResponse = self._client.ListVersions(request)
        return {"core": response.core, "api": response.api}
